src/platforms/esp32/main/main.c

```diff
@@ -76,12 +80,12 @@ void app_main()
     port_driver_init_all(glb);
     nif_collection_init_all(glb);
 
-    if (!avmpack_is_valid(main_avm, size)) {
-        ESP_LOGE(TAG, "Invalid main.avm packbeam.  size=%u", size);
+    if (!avmpack_is_valid(startup_avm, size)) {
+        ESP_LOGE(TAG, "Invalid startup avmpack. size=%u", size);
         AVM_ABORT();
     }
-    if (!avmpack_find_section_by_flag(main_avm, BEAM_START_FLAG, &startup_beam, &startup_beam_size, &startup_module_name)) {
-        ESP_LOGE(TAG, "Error: Failed to locate start module in main.avm packbeam.  (Did you flash a library by mistake?)");
+    if (!avmpack_find_section_by_flag(startup_avm, BEAM_START_FLAG, &startup_beam, &startup_beam_size, &startup_module_name)) {
```

Comment:
This will break images that are built with this change, until you
1.  Change the CMakeLists.txt file in the `libs` directory to build a pack beam file that contains the contents of `esp32.boot.avm` and `libatomvm.avm`
2. Change the mkimage.sh script to package this new AVM into the image

I believe in conversation you wanted to embed atomvmlib into esp32boot (or make a third AVM that contains both).  That work needs to be in this PR.

Reply:
Indeed `mkimage.sh` can be changed, however I'm pretty sure that this change shouldn't break current images, since it should behave as we did as fallback behavior.
Anyway it makes sense to do this further step.

Reply:
Steps to reproduce failure:

1. Download image from latest build on this branch https://github.com/atomvm/AtomVM/actions/runs/5692163785 (`atomvm-esp32-0.6.0-dev+git.1a7cdc0.img`)
2. Erase flash
```
> esptool.py --chip esp32 --port /dev/tty.usbserial-0001 --baud 921600 erase_flash
esptool.py v3.3.4-dev
Serial port /dev/tty.usbserial-0001
Connecting....
Chip is ESP32-D0WDQ6-V3 (revision v3.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 30:c6:f7:29:62:7c
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Erasing flash (this may take a while)...
Chip erase completed successfully in 5.3s
Hard resetting via RTS pin...
```

3. Flash image:
```
> esptool.py --chip esp32 --port /dev/tty.usbserial-0001 --baud 921600     --before default_reset --after hard_reset  write_flash -u --flash_mode dio --flash_freq 40m --flash_size detect     0x1000 ~/Downloads/atomvm-esp32-0.6.0-dev+git.1a7cdc0.img 
esptool.py v3.3.4-dev
Serial port /dev/tty.usbserial-0001
Connecting......
Chip is ESP32-D0WDQ6-V3 (revision v3.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 30:c6:f7:29:62:7c
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 921600
Changed.
Configuring flash size...
Auto-detected Flash size: 4MB
Flash will be erased from 0x00001000 to 0x001ebfff...
Wrote 2015232 bytes at 0x00001000 in 25.4 seconds (635.8 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
```

4. Observe crash in monitor:
```
> idf.py monitor -p /dev/tty.usbserial-0001
...
I (837) AtomVM: Starting AtomVM revision 0.6.0-dev+git.1a7cdc0
I (847) sys: Loaded BEAM partition boot.avm at address 0x1d0000 (size=262144 bytes)
I (867) network_driver: Initialized network interface
I (867) network_driver: Created default event loop
E (877) AtomVM: Error: Failed to locate start module in startup partition. (Did you flash a library by mistake?)

abort() was called at PC 0x400d82af on core 0
```